src/istore.c

```diff
@@ -976,3 +976,53 @@ istore_svals(PG_FUNCTION_ARGS)
 
     SRF_RETURN_DONE(funcctx);
 }
+
+/*
+ * A boolean function that supports `select '1=>1'::istore = '2=>2'::istore`
+ */
+PG_FUNCTION_INFO_V1(istore_eq);
+Datum
+istore_eq(PG_FUNCTION_ARGS)
+{
+    IStore *is1, *is2;
+    IStorePair *pairs1, *pairs2;
+    int32 len;
+
+    if (PG_ARGISNULL(0) || PG_ARGISNULL(1)) PG_RETURN_NULL();
+
+    is1 = PG_GETARG_IS(0);
+    is2 = PG_GETARG_IS(1);
+
+    if (is1->len != is2->len) PG_RETURN_BOOL(false);
+    len = is1->len;
+
+    pairs1 = FIRST_PAIR(is1, IStorePair);
+    pairs2 = FIRST_PAIR(is2, IStorePair);
+
+    PG_RETURN_BOOL(memcmp(pairs1, pairs2, sizeof(IStorePair) * len) == 0);
+}
+
+/*
+ * A boolean function that supports `select '1=>1'::bigistore = '2=>2'::bigistore`
+ */
+PG_FUNCTION_INFO_V1(bigistore_eq);
+Datum
+bigistore_eq(PG_FUNCTION_ARGS)
+{
+    BigIStore *is1, *is2;
+    BigIStorePair *pairs1, *pairs2;
+    int32 len;
+
+    if (PG_ARGISNULL(0) || PG_ARGISNULL(1)) PG_RETURN_NULL();
+
+    is1 = PG_GETARG_BIGIS(0);
+    is2 = PG_GETARG_BIGIS(1);
+
+    if (is1->len != is2->len) PG_RETURN_BOOL(false);
+    len = is1->len;
+
+    pairs1 = FIRST_PAIR(is1, BigIStorePair);
+    pairs2 = FIRST_PAIR(is2, BigIStorePair);
+
+    PG_RETURN_BOOL(memcmp(pairs1, pairs2, sizeof(BigIStorePair) * len) == 0);
```

Comment:
yep `BigIStorePair` is 16 byte and thus a `memcmp` won't work as we have 4 undefined bytes.